demo/src/contact-force-events/ContactForceEventsExample.tsx

```diff
@@ -0,0 +1,55 @@
+import { Plane, Sphere } from "@react-three/drei";
+import { MeshPhysicalMaterialProps } from "@react-three/fiber";
+import { RigidBody, RigidBodyApi, RigidBodyProps } from "@react-three/rapier";
+import { useRef, useState } from "react";
+import { Color } from "three";
+import { Demo } from "../App";
+
+
+type BallProps = { onContactForce: RigidBodyProps['onContactForce'] };
+const Ball = ({ onContactForce }: BallProps) => {
+  const ball = useRef<RigidBodyApi>(null);
+
+  return (
+    <RigidBody
+      ref={ball}
+      colliders="ball"
+      position={[0, 15, 0]}
+      restitution={0.5}
+      onContactForce={(payload) => {
+        const { totalForceMagnitude } = payload;
+        if (totalForceMagnitude < 300) {
+          ball.current?.applyImpulse({x: 0, y: 65, z: 0})
+        }
+        onContactForce?.(payload);
+      }}
+    >
+      <Sphere castShadow receiveShadow>
+        <meshPhysicalMaterial color="red" />
+      </Sphere>
+    </RigidBody>
+  );
+};
+type FloorProps = { color: MeshPhysicalMaterialProps['color']};
+const Floor = ({ color }: FloorProps) => {
+  return <RigidBody colliders="trimesh" type="fixed" restitution={1}>
+    <Plane args={[10, 10]} rotation={[Math.PI * 1.5, 0, 0]}>
+      <meshPhysicalMaterial color={color} />
+    </Plane>
+  </RigidBody>
+}
+
+const startColor = new Color(0x00ff00);
+export const ContactForceEventsExample: Demo = () => {
+  const [floorColor, setFloorColor] = useState(startColor.getHex());
+
+  return (
+    <group position={[0, -10, -10]}>
+      <Ball onContactForce={({ totalForceMagnitude }) => {
+        const color = startColor.clone().multiplyScalar(totalForceMagnitude / 6500);
+        setFloorColor(color.getHex());
```

Comment:
I couldn't think of a good way to demo this without adding an mp3 file and adjusting the volume of each hit, so I went with the color darkening as the hit becomes weaker. 6500 is a bit of a magic number here - its the force of the hit from the height the ball starts at.

Reply:
I kinda fixed the magic number. I moved the number up to a const and added a bit of a description as to why its there. I'm assuming its no big deal since its just an example.

I switched the colour effect too. It starts dark and gets lighter as the contacts get lighter.